[Bug]: Error: Plugin did not respond

### Terraform Core Version

1.1.7

### AWS Provider Version

4.55.0

### Affected Resource(s)

aws_cognito_risk_configuration

### Expected Behavior

Able to apply to config aws_cognito_risk_configuration

### Actual Behavior

Gives error as below

### Relevant Error/Panic Output Snippet

```shell
Stack trace from the terraform-provider-aws_v4.55.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 356 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/cognitoidp.expandRiskExceptionConfiguration({0xc003e6d620?, 0xdbc5711?, 0x0?})
        github.com/hashicorp/terraform-provider-aws/internal/service/cognitoidp/risk_configuration.go:384 +0x177
github.com/hashicorp/terraform-provider-aws/internal/service/cognitoidp.resourceRiskConfigurationPut({0xef1e2e0, 0xc003bd9860}, 0x0?, {0xdaa2c00?, 0xc000659c00?})
        github.com/hashicorp/terraform-provider-aws/internal/service/cognitoidp/risk_configuration.go:292 +0x2d1
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xef1e2e0?, {0xef1e2e0?, 0xc003bd9860?}, 0xd?, {0xdaa2c00?, 0xc000659c00?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:702 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000b510a0, {0xef1e2e0, 0xc003bd9860}, 0xc005bccc30, 0xc005bc4500, {0xdaa2c00, 0xc000659c00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000127470, {0xef1e2e0?, 0xc003bd9050?}, 0xc0056b3a40)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc00308ac60, 0xc00308acc0, {0xc0047d32a0, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001dc8320, {0xef1e2e0?, 0xc0030dde90?}, 0xc0056c5340)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xd7bf9c0?, 0xc001dc8320}, {0xef1e2e0, 0xc0030dde90}, 0xc0056c52d0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00404fe00, {0xef2d560, 0xc0048bf040}, 0xc005bbd0e0, 0xc004833d10, 0x1555db40, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc00404fe00, {0xef2d560, 0xc0048bf040}, 0xc005bbd0e0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-aws_v4.55.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

``` terraform
resource "aws_cognito_risk_configuration" "risk_config" {
  user_pool_id = aws_cognito_user_pool.pool.id
  client_id    = var.client_id

  account_takeover_risk_configuration {
    notify_configuration {
      dynamic block_email {
        for_each = var.block_email != null ? [1] : []

        content {
          html_body = lookup(var.block_email, "html_body")
          subject   = lookup(var.block_email, "subject")
          text_body = lookup(var.block_email, "text_body")
        }
      }
      dynamic mfa_email {
        for_each = var.mfa_email != null ? [1] : []

        content {
          html_body = lookup(var.mfa_email, "html_body")
          subject   = lookup(var.mfa_email, "subject")
          text_body = lookup(var.mfa_email, "text_body")
        }
      }
      dynamic no_action_email {
        for_each = var.no_action_email != null ? [1] : []
        
        content {
          html_body = lookup(var.no_action_email, "html_body")
          subject   = lookup(var.no_action_email, "subject")
          text_body = lookup(var.no_action_email, "text_body")
        }
      }

      from     = var.from
      reply_to = var.reply_to
      source_arn = "arn:aws:cognito-idp:ap-southeast-1:829891530814:userpool/ap-southeast-1_5CpSmHS2n"
    }
    actions {
      dynamic high_action {
        for_each = var.high_action != null ? [1] : []

        content {
          event_action = lookup(var.high_action, "event_action")
          notify       = lookup(var.high_action, "notify")
        }
      }

      dynamic medium_action {
        for_each = var.medium_action != null ? [1] : []

        content {
          event_action = lookup(var.medium_action, "event_action")
          notify       = lookup(var.medium_action, "notify")
        }
      }

      dynamic low_action {
        for_each = var.low_action != null ? [1] : []

        content {
          event_action = lookup(var.low_action, "event_action")
          notify       = lookup(var.low_action, "notify")
        }
      }

    }
  }

  compromised_credentials_risk_configuration {
    event_filter = var.event_filter
    actions {
      event_action = var.event_action
    }
  }

  risk_exception_configuration {
    blocked_ip_range_list = var.block_ip
    skipped_ip_range_list = var.skipped_ip
  }

  depends_on = [aws_cognito_user_pool.pool]
}

### Steps to Reproduce

Am using terragrunt.

terragrunt plan - no issue
terragrunt apply

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/30074.txt
+++ b/.changelog/30074.txt
@@ -1,0 +1,3 @@
+```release-note:bug    
+resource/aws_cognito_risk_configuration: Adds validation to `risk_exception_configuration` and requires at least one of `account_takeover_risk_configuration`, `compromised_credentials_risk_configuration`, or `risk_exception_configuration`.
+```

--- a/internal/acctest/acctest.go
+++ b/internal/acctest/acctest.go
@@ -2294,5 +2294,9 @@ func modulePrimaryInstanceState(ms *terraform.ModuleState, name string) (*terraf
 }
 
 func ExpectErrorAttrAtLeastOneOf(attrs ...string) *regexp.Regexp {
-	return regexp.MustCompile(fmt.Sprintf("one of[ \\n]`%s`[ \\n]must be specified", strings.Join(attrs, ",")))
+	return regexp.MustCompile(fmt.Sprintf("one of\\s+`%s`\\s+must be specified", strings.Join(attrs, ",")))
+}
+
+func ExpectErrorAttrMinItems(attr string, expected, actual int) *regexp.Regexp {
+	return regexp.MustCompile(fmt.Sprintf(`Attribute %s requires %d\s+item minimum, but config has only %d declared`, attr, expected, actual))
 }

--- a/internal/acctest/acctest.go
+++ b/internal/acctest/acctest.go
@@ -2292,3 +2292,7 @@ func modulePrimaryInstanceState(ms *terraform.ModuleState, name string) (*terraf
 
 	return is, nil
 }
+
+func ExpectErrorAttrAtLeastOneOf(attrs ...string) *regexp.Regexp {
+	return regexp.MustCompile(fmt.Sprintf("one of[ \\n]`%s`[ \\n]must be specified", strings.Join(attrs, ",")))
+}

--- a/internal/service/cognitoidp/risk_configuration.go
+++ b/internal/service/cognitoidp/risk_configuration.go
@@ -7,7 +7,6 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
-	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
@@ -15,6 +14,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/internal/create"
 	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
 	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
 	"github.com/hashicorp/terraform-provider-aws/internal/verify"
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
@@ -323,7 +323,7 @@ func resourceRiskConfigurationRead(ctx context.Context, d *schema.ResourceData, 
 
 	riskConfig, err := FindRiskConfigurationById(ctx, conn, d.Id())
 
-	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, cognitoidentityprovider.ErrCodeResourceNotFoundException) {
+	if !d.IsNewResource() && tfresource.NotFound(err) {
 		create.LogNotFoundRemoveState(names.CognitoIDP, create.ErrActionReading, ResNameRiskConfiguration, d.Id())
 		d.SetId("")
 		return diags

--- a/internal/service/cognitoidp/risk_configuration.go
+++ b/internal/service/cognitoidp/risk_configuration.go
@@ -249,10 +249,10 @@ func ResourceRiskConfiguration() *schema.Resource {
 						"blocked_ip_range_list": {
 							Type:     schema.TypeSet,
 							Optional: true,
+							MaxItems: 200,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 								ValidateFunc: validation.All(
-									validation.StringLenBetween(0, 200),
 									validation.IsCIDR,
 								),
 							},
@@ -260,10 +260,10 @@ func ResourceRiskConfiguration() *schema.Resource {
 						"skipped_ip_range_list": {
 							Type:     schema.TypeSet,
 							Optional: true,
+							MaxItems: 200,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 								ValidateFunc: validation.All(
-									validation.StringLenBetween(0, 200),
 									validation.IsCIDR,
 								)},
 						},

--- a/internal/service/cognitoidp/risk_configuration.go
+++ b/internal/service/cognitoidp/risk_configuration.go
@@ -46,6 +46,11 @@ func ResourceRiskConfiguration() *schema.Resource {
 				Type:     schema.TypeList,
 				Optional: true,
 				MaxItems: 1,
+				AtLeastOneOf: []string{
+					"account_takeover_risk_configuration",
+					"compromised_credentials_risk_configuration",
+					"risk_exception_configuration",
+				},
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"actions": {

--- a/internal/service/cognitoidp/risk_configuration.go
+++ b/internal/service/cognitoidp/risk_configuration.go
@@ -254,7 +254,12 @@ func ResourceRiskConfiguration() *schema.Resource {
 						"blocked_ip_range_list": {
 							Type:     schema.TypeSet,
 							Optional: true,
+							MinItems: 1,
 							MaxItems: 200,
+							AtLeastOneOf: []string{
+								"risk_exception_configuration.0.blocked_ip_range_list",
+								"risk_exception_configuration.0.skipped_ip_range_list",
+							},
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 								ValidateFunc: validation.All(
@@ -265,6 +270,7 @@ func ResourceRiskConfiguration() *schema.Resource {
 						"skipped_ip_range_list": {
 							Type:     schema.TypeSet,
 							Optional: true,
+							MinItems: 1,
 							MaxItems: 200,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
@@ -387,6 +393,10 @@ func resourceRiskConfigurationDelete(ctx context.Context, d *schema.ResourceData
 }
 
 func expandRiskExceptionConfiguration(riskConfig []interface{}) *cognitoidentityprovider.RiskExceptionConfigurationType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	riskExceptionConfigurationType := &cognitoidentityprovider.RiskExceptionConfigurationType{}
@@ -421,6 +431,10 @@ func flattenRiskExceptionConfiguration(apiObject *cognitoidentityprovider.RiskEx
 }
 
 func expandCompromisedCredentialsRiskConfiguration(riskConfig []interface{}) *cognitoidentityprovider.CompromisedCredentialsRiskConfigurationType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	riskExceptionConfigurationType := &cognitoidentityprovider.CompromisedCredentialsRiskConfigurationType{}
@@ -455,6 +469,10 @@ func flattenCompromisedCredentialsRiskConfiguration(apiObject *cognitoidentitypr
 }
 
 func expandCompromisedCredentialsActions(riskConfig []interface{}) *cognitoidentityprovider.CompromisedCredentialsActionsType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	compromisedCredentialsAction := &cognitoidentityprovider.CompromisedCredentialsActionsType{}
@@ -481,6 +499,10 @@ func flattenCompromisedCredentialsActions(apiObject *cognitoidentityprovider.Com
 }
 
 func expandAccountTakeoverRiskConfiguration(riskConfig []interface{}) *cognitoidentityprovider.AccountTakeoverRiskConfigurationType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	accountTakeoverRiskConfiguration := &cognitoidentityprovider.AccountTakeoverRiskConfigurationType{}
@@ -515,6 +537,10 @@ func flattenAccountTakeoverRiskConfiguration(apiObject *cognitoidentityprovider.
 }
 
 func expandAccountTakeoverActions(riskConfig []interface{}) *cognitoidentityprovider.AccountTakeoverActionsType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	actions := &cognitoidentityprovider.AccountTakeoverActionsType{}
@@ -557,6 +583,10 @@ func flattenAccountTakeoverActions(apiObject *cognitoidentityprovider.AccountTak
 }
 
 func expandAccountTakeoverAction(riskConfig []interface{}) *cognitoidentityprovider.AccountTakeoverActionType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	action := &cognitoidentityprovider.AccountTakeoverActionType{}
@@ -591,6 +621,10 @@ func flattenAccountTakeoverAction(apiObject *cognitoidentityprovider.AccountTake
 }
 
 func expandNotifyConfiguration(riskConfig []interface{}) *cognitoidentityprovider.NotifyConfigurationType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	notifConfig := &cognitoidentityprovider.NotifyConfigurationType{}
@@ -657,6 +691,10 @@ func flattenNotifyConfiguration(apiObject *cognitoidentityprovider.NotifyConfigu
 }
 
 func expandNotifyEmail(riskConfig []interface{}) *cognitoidentityprovider.NotifyEmailType {
+	if len(riskConfig) == 0 || riskConfig[0] == nil {
+		return nil
+	}
+
 	config := riskConfig[0].(map[string]interface{})
 
 	notifyEmail := &cognitoidentityprovider.NotifyEmailType{}

--- a/internal/service/cognitoidp/risk_configuration_test.go
+++ b/internal/service/cognitoidp/risk_configuration_test.go
@@ -140,7 +140,7 @@ func TestAccCognitoIDPRiskConfiguration_disappears(t *testing.T) {
 	resourceName := "aws_cognito_risk_configuration.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:                 func() { acctest.PreCheck(t); testAccPreCheckIdentityProvider(ctx, t) },
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheckIdentityProvider(ctx, t) },
 		ErrorCheck:               acctest.ErrorCheck(t, cognitoidentityprovider.EndpointsID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
 		CheckDestroy:             testAccCheckRiskConfigurationDestroy(ctx),
@@ -185,7 +185,7 @@ func TestAccCognitoIDPRiskConfiguration_empty(t *testing.T) {
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:                 func() { acctest.PreCheck(t); testAccPreCheckIdentityProvider(ctx, t) },
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheckIdentityProvider(ctx, t) },
 		ErrorCheck:               acctest.ErrorCheck(t, cognitoidentityprovider.EndpointsID),
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
 		CheckDestroy:             testAccCheckRiskConfigurationDestroy(ctx),
@@ -197,6 +197,45 @@ func TestAccCognitoIDPRiskConfiguration_empty(t *testing.T) {
 					"compromised_credentials_risk_configuration",
 					"risk_exception_configuration",
 				),
+			},
+		},
+	})
+}
+
+func TestAccCognitoIDPRiskConfiguration_nullRiskException(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheckIdentityProvider(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, cognitoidentityprovider.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRiskConfigurationDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRiskConfigurationConfig_nullRiskException(rName),
+				ExpectError: acctest.ExpectErrorAttrAtLeastOneOf(
+					"risk_exception_configuration.0.blocked_ip_range_list",
+					"risk_exception_configuration.0.skipped_ip_range_list",
+				),
+			},
+		},
+	})
+}
+
+func TestAccCognitoIDPRiskConfiguration_emptyRiskException(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheckIdentityProvider(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, cognitoidentityprovider.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRiskConfigurationDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccRiskConfigurationConfig_emptyRiskException(rName),
+				ExpectError: acctest.ExpectErrorAttrMinItems("risk_exception_configuration.0.blocked_ip_range_list", 1, 0),
 			},
 		},
 	})
@@ -324,6 +363,40 @@ func testAccRiskConfigurationConfig_empty(rName string) string {
 	return fmt.Sprintf(`
 resource "aws_cognito_risk_configuration" "test" {
   user_pool_id = aws_cognito_user_pool.test.id
+}
+
+resource "aws_cognito_user_pool" "test" {
+  name = %[1]q
+}
+`, rName)
+}
+
+func testAccRiskConfigurationConfig_nullRiskException(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_cognito_risk_configuration" "test" {
+  user_pool_id = aws_cognito_user_pool.test.id
+
+  risk_exception_configuration {
+    blocked_ip_range_list = null
+    skipped_ip_range_list = null
+  }
+}
+
+resource "aws_cognito_user_pool" "test" {
+  name = %[1]q
+}
+`, rName)
+}
+
+func testAccRiskConfigurationConfig_emptyRiskException(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_cognito_risk_configuration" "test" {
+  user_pool_id = aws_cognito_user_pool.test.id
+
+  risk_exception_configuration {
+    blocked_ip_range_list = []
+    skipped_ip_range_list = []
+  }
 }
 
 resource "aws_cognito_user_pool" "test" {

--- a/internal/service/cognitoidp/risk_configuration_test.go
+++ b/internal/service/cognitoidp/risk_configuration_test.go
@@ -180,6 +180,28 @@ func TestAccCognitoIDPRiskConfiguration_disappears_userPool(t *testing.T) {
 	})
 }
 
+func TestAccCognitoIDPRiskConfiguration_empty(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t); testAccPreCheckIdentityProvider(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, cognitoidentityprovider.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRiskConfigurationDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRiskConfigurationConfig_empty(rName),
+				ExpectError: acctest.ExpectErrorAttrAtLeastOneOf(
+					"account_takeover_risk_configuration",
+					"compromised_credentials_risk_configuration",
+					"risk_exception_configuration",
+				),
+			},
+		},
+	})
+}
+
 func testAccCheckRiskConfigurationDestroy(ctx context.Context) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		conn := acctest.Provider.Meta().(*conns.AWSClient).CognitoIDPConn()
@@ -294,6 +316,18 @@ resource "aws_cognito_user_pool_client" "test" {
   name                = %[1]q
   user_pool_id        = aws_cognito_user_pool.test.id
   explicit_auth_flows = ["ADMIN_NO_SRP_AUTH"]
+}
+`, rName)
+}
+
+func testAccRiskConfigurationConfig_empty(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_cognito_risk_configuration" "test" {
+  user_pool_id = aws_cognito_user_pool.test.id
+}
+
+resource "aws_cognito_user_pool" "test" {
+  name = %[1]q
 }
 `, rName)
 }

--- a/internal/service/cognitoidp/risk_configuration_test.go
+++ b/internal/service/cognitoidp/risk_configuration_test.go
@@ -134,6 +134,29 @@ func TestAccCognitoIDPRiskConfiguration_compromised(t *testing.T) {
 	})
 }
 
+func TestAccCognitoIDPRiskConfiguration_disappears(t *testing.T) {
+	ctx := acctest.Context(t)
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_cognito_risk_configuration.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t); testAccPreCheckIdentityProvider(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, cognitoidentityprovider.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckRiskConfigurationDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRiskConfigurationConfig_riskException(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckRiskConfigurationExists(ctx, resourceName),
+					acctest.CheckResourceDisappears(ctx, acctest.Provider, tfcognitoidp.ResourceRiskConfiguration(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
 func TestAccCognitoIDPRiskConfiguration_disappears_userPool(t *testing.T) {
 	ctx := acctest.Context(t)
 	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)

--- a/internal/service/cognitoidp/risk_configuration_test.go
+++ b/internal/service/cognitoidp/risk_configuration_test.go
@@ -32,12 +32,13 @@ func TestAccCognitoIDPRiskConfiguration_exception(t *testing.T) {
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckRiskConfigurationExists(ctx, resourceName),
 					resource.TestCheckResourceAttrPair(resourceName, "user_pool_id", "aws_cognito_user_pool.test", "id"),
+					resource.TestCheckNoResourceAttr(resourceName, "client_id"),
+					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.#", "1"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.#", "1"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.*", "10.10.10.10/32"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.0.skipped_ip_range_list.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
 				),
 			},
 			{
@@ -50,14 +51,15 @@ func TestAccCognitoIDPRiskConfiguration_exception(t *testing.T) {
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckRiskConfigurationExists(ctx, resourceName),
 					resource.TestCheckResourceAttrPair(resourceName, "user_pool_id", "aws_cognito_user_pool.test", "id"),
+					resource.TestCheckNoResourceAttr(resourceName, "client_id"),
+					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.#", "1"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.#", "2"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.*", "10.10.10.10/32"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.*", "10.10.10.11/32"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.0.skipped_ip_range_list.#", "1"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "risk_exception_configuration.0.skipped_ip_range_list.*", "10.10.10.12/32"),
-					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
 				),
 			},
 		},
@@ -81,12 +83,12 @@ func TestAccCognitoIDPRiskConfiguration_client(t *testing.T) {
 					testAccCheckRiskConfigurationExists(ctx, resourceName),
 					resource.TestCheckResourceAttrPair(resourceName, "user_pool_id", "aws_cognito_user_pool.test", "id"),
 					resource.TestCheckResourceAttrPair(resourceName, "client_id", "aws_cognito_user_pool_client.test", "id"),
+					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.#", "1"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.#", "1"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "risk_exception_configuration.0.blocked_ip_range_list.*", "10.10.10.10/32"),
 					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.0.skipped_ip_range_list.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "0"),
-					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
 				),
 			},
 			{
@@ -114,13 +116,13 @@ func TestAccCognitoIDPRiskConfiguration_compromised(t *testing.T) {
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckRiskConfigurationExists(ctx, resourceName),
 					resource.TestCheckResourceAttrPair(resourceName, "user_pool_id", "aws_cognito_user_pool.test", "id"),
-					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
 					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.#", "1"),
 					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.0.event_filter.#", "1"),
 					resource.TestCheckTypeSetElemAttr(resourceName, "compromised_credentials_risk_configuration.0.event_filter.*", "SIGN_IN"),
 					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.0.actions.#", "1"),
 					resource.TestCheckResourceAttr(resourceName, "compromised_credentials_risk_configuration.0.actions.0.event_action", "BLOCK"),
-					resource.TestCheckResourceAttr(resourceName, "account_takeover_risk_configuration.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "risk_exception_configuration.#", "0"),
 				),
 			},
 			{
@@ -200,10 +202,6 @@ func testAccCheckRiskConfigurationExists(ctx context.Context, name string) resou
 
 func testAccRiskConfigurationConfig_riskException(rName string) string {
 	return fmt.Sprintf(`
-resource "aws_cognito_user_pool" "test" {
-  name = %[1]q
-}
-
 resource "aws_cognito_risk_configuration" "test" {
   user_pool_id = aws_cognito_user_pool.test.id
 
@@ -211,15 +209,15 @@ resource "aws_cognito_risk_configuration" "test" {
     blocked_ip_range_list = ["10.10.10.10/32"]
   }
 }
+
+resource "aws_cognito_user_pool" "test" {
+  name = %[1]q
+}
 `, rName)
 }
 
 func testAccRiskConfigurationConfig_riskExceptionUpdated(rName string) string {
 	return fmt.Sprintf(`
-resource "aws_cognito_user_pool" "test" {
-  name = %[1]q
-}
-
 resource "aws_cognito_risk_configuration" "test" {
   user_pool_id = aws_cognito_user_pool.test.id
 
@@ -228,15 +226,15 @@ resource "aws_cognito_risk_configuration" "test" {
     skipped_ip_range_list = ["10.10.10.12/32"]
   }
 }
+
+resource "aws_cognito_user_pool" "test" {
+  name = %[1]q
+}
 `, rName)
 }
 
 func testAccRiskConfigurationConfig_compromised(rName string) string {
 	return fmt.Sprintf(`
-resource "aws_cognito_user_pool" "test" {
-  name = %[1]q
-}
-
 resource "aws_cognito_risk_configuration" "test" {
   user_pool_id = aws_cognito_user_pool.test.id
 
@@ -247,11 +245,24 @@ resource "aws_cognito_risk_configuration" "test" {
     }
   }
 }
+
+resource "aws_cognito_user_pool" "test" {
+  name = %[1]q
+}
 `, rName)
 }
 
 func testAccRiskConfigurationConfig_riskExceptionClient(rName string) string {
 	return fmt.Sprintf(`
+resource "aws_cognito_risk_configuration" "test" {
+  user_pool_id = aws_cognito_user_pool.test.id
+  client_id    = aws_cognito_user_pool_client.test.id
+
+  risk_exception_configuration {
+    blocked_ip_range_list = ["10.10.10.10/32"]
+  }
+}
+
 resource "aws_cognito_user_pool" "test" {
   name = %[1]q
 }
@@ -260,15 +271,6 @@ resource "aws_cognito_user_pool_client" "test" {
   name                = %[1]q
   user_pool_id        = aws_cognito_user_pool.test.id
   explicit_auth_flows = ["ADMIN_NO_SRP_AUTH"]
-}
-
-resource "aws_cognito_risk_configuration" "test" {
-  user_pool_id = aws_cognito_user_pool.test.id
-  client_id    = aws_cognito_user_pool_client.test.id
-
-  risk_exception_configuration {
-    blocked_ip_range_list = ["10.10.10.10/32"]
-  }
 }
 `, rName)
 }

--- a/website/docs/r/cognito_risk_configuration.html.markdown
+++ b/website/docs/r/cognito_risk_configuration.html.markdown
@@ -74,8 +74,12 @@ The following arguments are supported:
 
 ### risk_exception_configuration
 
-* `blocked_ip_range_list` - (Optional) Overrides the risk decision to always block the pre-authentication requests. The IP range is in CIDR notation, a compact representation of an IP address and its routing prefix.
-* `skipped_ip_range_list` - (Optional) Risk detection isn't performed on the IP addresses in this range list. The IP range is in CIDR notation.
+* `blocked_ip_range_list` - (Optional) Overrides the risk decision to always block the pre-authentication requests.
+  The IP range is in CIDR notation, a compact representation of an IP address and its routing prefix.
+  Can contain a maximum of 200 items.
+* `skipped_ip_range_list` - (Optional) Risk detection isn't performed on the IP addresses in this range list.
+  The IP range is in CIDR notation.
+  Can contain a maximum of 200 items.
 
 ## Attributes Reference
 

